### PR TITLE
[No QA] Card settings page

### PR DESCRIPTION
**Pull request by koko57**
### Fixed Issues

$ https://github.com/Expensify/App/issues/44311
$ https://github.com/Expensify/App/issues/44312
$ https://github.com/Expensify/App/issues/44313
PROPOSAL: -

### Tests

PREREQUISITES: Comment out `AccessOrNotFoundWrapper` in `WorkspaceCardSettingsPage`, `WorkspaceSettlementAccountPage` and `WorkspaceSettlementFrequencyPage`

1. WEB/mWEB: Go to settings/workspaces/[yourWorkspace]. In the browser add expensify-card/settings to the URL. 
MOBILE: (with simulator open and project build) in the terminal type `npx uri-scheme open new-expensify://settings/workspaces/[workspaceId]/expensify-card/settings --ios` to open on iOS, `npx uri-scheme open new-expensify://settings/workspaces/[workspaceId]/expensify-card/settings --android` to open on android.
DESKTOP: with the desktop app open, in the browser type https://staging.new.expensify.com/settings/workspaces/[workspaceId]/expensify-card/settings, when the prompt to open the app will be displayed, open the app
2. Verify that the settings page displays correctly with two items.
3. Go to Frequency setting, verify that options are displayed correctly
4. Go to Account setting, verify that options are displayed correctly

🚨 NOTE: for now I'm not filtering out eligible accounts for easier testing. It will be done when integrating with the backend. I also haven't disabled Frequency menu item to make it possible to check both Daily and Monthly state

- [x] Verify that no errors appear in the JS console

### QA Steps

n/a

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

Screenshot_1721124990
Screenshot_1721124983
Screenshot_1721124996
Screenshot_1721124999

</details>

<details>
<summary>Android: mWeb Chrome</summary>

Screenshot_1721119210
Screenshot_1721119214
Screenshot_1721119207

</details>

<details>
<summary>iOS: Native</summary>

Simulator Screenshot - iPhone 15 Pro - 2024-07-16 at 11 47 12
Simulator Screenshot - iPhone 15 Pro - 2024-07-16 at 11 47 46
Simulator Screenshot - iPhone 15 Pro - 2024-07-16 at 11 48 12

</details>

<details>
<summary>iOS: mWeb Safari</summary>

Simulator Screenshot - iPhone 15 Pro - 2024-07-16 at 10 33 34
Simulator Screenshot - iPhone 15 Pro - 2024-07-16 at 10 33 31
Simulator Screenshot - iPhone 15 Pro - 2024-07-16 at 10 33 23

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

Screenshot 2024-07-16 at 09 57 53
Screenshot 2024-07-16 at 09 56 27
Screenshot 2024-07-16 at 09 56 10
Screenshot 2024-07-16 at 10 23 27

</details>

<details>
<summary>MacOS: Desktop</summary>

Screenshot 2024-07-16 at 10 28 55
Screenshot 2024-07-16 at 10 28 33
Screenshot 2024-07-16 at 10 27 33

</details>

### CODE REVIEW

File: src/components/SelectionList/RadioListItem.tsx [hunk 1]
```diff
@@ -41,6 +41,7 @@ function RadioListItem<TItem extends ListItem>({
             pendingAction={item.pendingAction}
         >
             <>
+                {!!item.leftElement && item.leftElement}
```
[the remark this answers is not in the document]
Comment (koko57) [reply src/components/SelectionList/RadioListItem.tsx:44]: If you would pass an empty string it would render an empty string. And in the code it looks consistent - look at the item.rightElement or item.alternateText

File: src/libs/CardUtils.ts [hunk 2]
```diff
@@ -158,6 +159,14 @@ function getTranslationKeyForLimitType(limitType: ValueOf<typeof CONST.EXPENSIFY
     }
 }
 
+function getEligibleBankAccountsForCard(bankAccountsList: OnyxEntry<BankAccountList>) {
+    if (!bankAccountsList || isEmptyObject(bankAccountsList)) {
+        return [];
+    }
+    /* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
+    return Object.values(bankAccountsList).filter((bankAccount) => bankAccount?.accountData?.allowDebit || bankAccount?.accountData?.type === CONST.BANK_ACCOUNT.TYPE.BUSINESS);
```
[the remark this answers is not in the document]
Comment (koko57) [reply src/libs/CardUtils.ts:167]: if allowDebit would be false it would return false and wouldn't check for the type. We could have false for allowDebit but type would be fine, but we would get false incorrectly

**Review by allgandalf — COMMENTED** [review]
@Expensify/design can you guys trigger an `ad-hoc build` and confirm the design please, I will review this from technical standpoint simultaneously, thanks.

**Comment by shawnborton** [comment]
Just a heads up @dannymcclain and @dubielzyk-expensify - I added you both to the beta, but it will take some time to populate in your accounts.

**Comment by koko57** [comment]
@dannymcclain for now it's not disabled (it should be but it'll be implemented later), so it's hoverable. I think it shouldn't be hoverable when it'll be disabled, am I right @shawnborton?

**Comment by shawnborton** [comment]
Good catch - hint text should definitely be within the hoverable area. I'm having a tough time finding an example of it, but I am pretty sure it exists somewhere in the accounting integration screens we built.

**Comment by shawnborton** [comment]
Ah, we wouldn't "disable" it but I think we would just make it read-only, right? At which case you are right, no hover styles to worry about. But if this will ever be interactive, let's make sure the hint text is included in the hoverable area.

**Comment by dannymcclain** [comment]
>Ah, we wouldn't "disable" it but I think we would just make it read-only, right? At which case you are right, no hover styles to worry about. But if this will ever be interactive, let's make sure the hint text is included in the hoverable area.

Ah yeah ok. Totally agree with everything you've said here.

**Comment by koko57** [comment]
@shawnborton @dannymcclain for now it's hoverable, bc it's not locked but now the hint text is inside the MenuItem

Screenshot 2024-07-17 at 09 11 56Screenshot 2024-07-17 at 09 12 04

**Comment by koko57** [comment]
@allgandalf ready for the another round of review

**Comment by koko57** [comment]
@allgandalf fixed!

**Review by allgandalf — COMMENTED** [review]
couple of comments, technically sounds good, I guess the design team still hasn't tested it out fully they have some [issue](https://expensify.slack.com/archives/C036QM0SLJK/p1721195479459249?thread_ts=1721057595.798109&cid=C036QM0SLJK) accessing the pages

**Review by allgandalf — COMMENTED** [review]
I am seeing repeated use of `StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.EXPENSIFY_CARD_SETTINGS>`, lets define a type for this under a new file and maybe import it on the pages where it is required, what do you think @koko57 ?

**Comment by koko57** [comment]
@allgandalf 

> I am seeing repeated use of `StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.WORKSPACE.EXPENSIFY_CARD_SETTINGS>`, lets define a type for this under a new file and maybe import it on the pages where it is required, what do you think @koko57 ?

@allgandalf actually my bad - these 3 pages should not have the same type, they have their own screen type

File: src/types/onyx/ExpensifyCardSettings.ts [hunk 3]
```diff
@@ -10,6 +10,15 @@ type ExpensifyCardSettings = OnyxCommon.OnyxValueWithOfflineFeedback<{
...
     cashBack: number;
+
+    /** The date of the last settlement */
+    monthlySettlementDate: Date;
+
+    /** Whether monthly option should appear in the settlement frequency settings */
+    isMonthlySettlementAllowed: boolean;
+
+    /** The bank account chosen for the card settlement */
+    paymentBankAccountID: number;
```
[the remark this answers is not in the document]
Comment (koko57) [reply src/types/onyx/ExpensifyCardSettings.ts:21]: @MariaHCD should it be a number or a string?

[the remark this answers is not in the document]
Comment (MariaHCD) [reply src/types/onyx/ExpensifyCardSettings.ts:21]: paymentBankAccountID is an integer 👍🏼

**Comment by shawnborton** [comment]
> for now it's hoverable, bc it's not locked but now the hint text is inside the MenuItem

If it can't be interacted with, can we make it so that it doesn't have any hover styles please? Basically the idea is that it would be read-only at that point.

**Comment by shawnborton** [comment]
> were you guys able to get through this ?

I think it will be easier for us to just review this based on screenshots for now.

**Comment by koko57** [comment]
@shawnborton yes it won't be hoverable - I just made it interactive only for testing purposes - otherwise we would'n be able to go to the frequency page. I planned to do that in the follow-up with the BE integration, but I can do it in this PR if you prefer

**Comment by shawnborton** [comment]
Cool cool, however you prefer to get there works for me - just want to make sure we handle it at some point. Thanks!

**Comment by koko57** [comment]
@allgandalf fixed

File: src/components/SelectionList/RadioListItem.tsx (hunk 1 above)
Comment (mountiny) [inline src/components/SelectionList/RadioListItem.tsx:44]: It would be amazing to make these changes in a separate PR with clear test steps. Its always tricky to make changes to the low-level components

Comment (mountiny) [reply src/components/SelectionList/RadioListItem.tsx:44]: but I assume this should be fine

File: src/libs/CardUtils.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mountiny) [reply src/libs/CardUtils.ts:167]: Could we get around this? I believe that any eslnt rule disable means that react-compiler does not process this file, we should try to avoid adding any eslint-disable to new code

File: src/pages/workspace/expensifyCard/WorkspaceSettlementAccountPage.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+            >
+                <HeaderWithBackButton title={translate('workspace.expensifyCard.settlementAccount')} />
+                <Text style={[styles.mh5, styles.mv4]}>{translate('workspace.expensifyCard.settlementAccountDescription')}</Text>
+                {isUsedContinuousReconciliation && (
+                    <Text style={[styles.mh5, styles.mb6]}>
+                        <Text>{translate('workspace.expensifyCard.settlementAccountInfoPt1')}</Text>{' '}
+                        <TextLink onPress={() => Navigation.navigate(ROUTES.WORKSPACE_ACCOUNTING_RECONCILIATION_ACCOUNT_SETTINGS.getRoute(policyID, reconciliationConnection))}>
+                            {translate('workspace.expensifyCard.reconciliationAccount')}
+                        </TextLink>{' '}
+                        <Text>{`(XXXXXXXXXXXX${getLastFourDigits(paymentBankAccountNumber)}) `}</Text>
```
Comment (mountiny) [inline src/pages/workspace/expensifyCard/WorkspaceSettlementAccountPage.tsx:102]: Could you make `XXXXXXXXXXXX` a const something like: `maskedPANPrefix`

File: src/pages/workspace/expensifyCard/WorkspaceSettlementFrequencyPage.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+function WorkspaceSettlementFrequencyPage({route}: WorkspaceSettlementFrequencyPageProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const policyID = route.params?.policyID ?? '-1';
+
+    const [cardSettings] = useOnyx(`${ONYXKEYS.COLLECTION.SHARED_NVP_PRIVATE_EXPENSIFY_CARD_SETTINGS}${policyID}`);
+
+    // TODO: change true for false after API is ready - true is for testing purposes
+    const showMonthlyOption = cardSettings?.isMonthlySettlementAllowed ?? true;
```
Comment (mountiny) [inline src/pages/workspace/expensifyCard/WorkspaceSettlementFrequencyPage.tsx:30]:
```suggestion
    const shouldShowMonthlyOption = cardSettings?.isMonthlySettlementAllowed ?? true;
```

File: src/pages/workspace/expensifyCard/WorkspaceSettlementFrequencyPage.tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        const options = [];
+
+        options.push({
+            value: CONST.EXPENSIFY_CARD.FREQUENCY_SETTING.DAILY,
+            text: translate('workspace.expensifyCard.frequency.daily'),
+            keyForList: CONST.EXPENSIFY_CARD.FREQUENCY_SETTING.DAILY,
+            isSelected: selectedFrequency === CONST.EXPENSIFY_CARD.FREQUENCY_SETTING.DAILY,
+        });
+
+        if (showMonthlyOption) {
```
Comment (mountiny) [inline src/pages/workspace/expensifyCard/WorkspaceSettlementFrequencyPage.tsx:43]:
```suggestion
        if (shouldShowMonthlyOption) {
```

File: src/pages/workspace/expensifyCard/WorkspaceSettlementFrequencyPage.tsx [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+            options.push({
+                value: CONST.EXPENSIFY_CARD.FREQUENCY_SETTING.MONTHLY,
+                text: translate('workspace.expensifyCard.frequency.monthly'),
+                keyForList: CONST.EXPENSIFY_CARD.FREQUENCY_SETTING.MONTHLY,
+                isSelected: selectedFrequency === CONST.EXPENSIFY_CARD.FREQUENCY_SETTING.MONTHLY,
+            });
+        }
+
+        return options;
+    }, [translate, showMonthlyOption, selectedFrequency]);
```
Comment (mountiny) [inline src/pages/workspace/expensifyCard/WorkspaceSettlementFrequencyPage.tsx:53]:
```suggestion
    }, [translate, shouldShowMonthlyOption, selectedFrequency]);
```

**Review by mountiny — COMMENTED** [review]
@koko57 looking great, @allgandalf  lets try to wrap up the review here

File: src/components/SelectionList/RadioListItem.tsx (hunk 1 above)
Comment (koko57) [reply src/components/SelectionList/RadioListItem.tsx:44]: yes, as it's an optional property this shouldn't cause any problem, I also haven't noticed anything off with this component so far

File: src/ONYXKEYS.ts [hunk 8]
```diff
@@ -449,10 +449,10 @@ const ONYXKEYS = {
          */
         WORKSPACE_CARDS_LIST: 'card_',
 
-        /** The bank account that Expensify Card payments will be reconciled against */
+        /** Stores which connection is set up to use Continuous Reconciliation */
         SHARED_NVP_EXPENSIFY_CARD_CONTINUOUS_RECONCILIATION_CONNECTION: 'sharedNVP_expensifyCard_continuousReconciliationConnection_',
```
Comment (koko57) [inline src/ONYXKEYS.ts:453]: @MariaHCD @mountiny - just to make sure, sharedNVP_expensifyCard_continuousReconciliationConnection_ should store connection name, not the bank account data?

Comment (MariaHCD) [reply src/ONYXKEYS.ts:453]: Yes, sharedNVP_expensifyCard_continuousReconciliationConnection_ will just store the name of the connection like `quickbooksOnline`, `xero`, `netsuite` or `intacct`

Comment (koko57) [reply src/ONYXKEYS.ts:453]: ok, thanks! So it's now fixed in this PR

**Comment by koko57** [comment]
@allgandalf fixed! 😃

File: src/pages/workspace/expensifyCard/WorkspaceSettlementAccountPage.tsx [hunk 9]
```diff
@@ -0,0 +1,120 @@
+import type {StackScreenProps} from '@react-navigation/stack';
```
[the remark this answers is not in the document]
Comment (koko57) [reply src/pages/workspace/expensifyCard/WorkspaceSettlementAccountPage.tsx:1]: you have to have a bank account added (through the VBBA flow)

File: src/ONYXKEYS.ts (hunk 8 above)
Comment (MariaHCD) [reply src/ONYXKEYS.ts:453]: It will just be a string. I believe it's only one connection at a time.

**Comment by koko57** [comment]
@allgandalf fixed the navigation

File: src/pages/workspace/expensifyCard/WorkspaceSettlementAccountPage.tsx [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+            policyID={policyID}
+            featureName={CONST.POLICY.MORE_FEATURES.ARE_EXPENSIFY_CARDS_ENABLED}
+        >
+            <ScreenWrapper
+                testID={WorkspaceSettlementAccountPage.displayName}
+                includeSafeAreaPaddingBottom={false}
+                shouldEnableMaxHeight
+            >
+                <HeaderWithBackButton title={translate('workspace.expensifyCard.settlementAccount')} />
+                <Text style={[styles.mh5, styles.mv4]}>{translate('workspace.expensifyCard.settlementAccountDescription')}</Text>
```
[the remark this answers is not in the document]
Comment (koko57) [reply src/pages/workspace/expensifyCard/WorkspaceSettlementAccountPage.tsx]: applied!

**Comment by koko57** [comment]
> ### Bug
> On frequency page, the highlight doesn't move to the newly selected value
> 
>  Screen.Recording.2024-07-19.at.3.44.39.PM.mov

@allgandalf - this is something that has not been implemented for this component. @VickyStash added it to https://github.com/Expensify/App/pull/45601/files, we can incorporate these changes in a follow-up (when connecting to the BE)

**Review by allgandalf — APPROVED** [review]
This tests well on all platforms, the bug i found was mentioned [here](https://github.com/Expensify/App/pull/45415#issuecomment-2238882613), would actually be solved when we implement the `BE`, I will keep an eye out because that bug is very much a blocker to me if i cannot shift the focus on options, rest assured, we can merge this one before the week closes, thanks for the back and forth @koko57!

**Comment by koko57** [comment]
> Screenshot 2024-07-19 at 5 45 20 PM
> NAB: When i click on learn more nothing happens really, is it for future implementation @koko57 ?

yep, the doc is being worked on, once it's done we'll get a proper url to link to

**Comment by trjExpensify** [comment]
Great! It would be awesome to get this merged today, as it's the last PR left in Phase 2. 🎉

**Review by mountiny — DISMISSED** [review]
Changes look good to me, thank you @koko57 and @allgandalf for quick testing.

This is now all ready for @MariaHCD

File: src/libs/CardUtils.ts [hunk 11]
```diff
@@ -158,6 +159,13 @@ function getTranslationKeyForLimitType(limitType: ValueOf<typeof CONST.EXPENSIFY
     }
 }
 
+function getEligibleBankAccountsForCard(bankAccountsList: OnyxEntry<BankAccountList>) {
+    if (!bankAccountsList || isEmptyObject(bankAccountsList)) {
+        return [];
+    }
+    return Object.values(bankAccountsList).filter((bankAccount) => bankAccount?.accountData?.type === CONST.BANK_ACCOUNT.TYPE.BUSINESS || bankAccount?.accountData?.allowDebit);
```
Comment (MariaHCD) [inline src/libs/CardUtils.ts:166]:
```suggestion
    return Object.values(bankAccountsList).filter((bankAccount) => bankAccount?.accountData?.type === CONST.BANK_ACCOUNT.TYPE.BUSINESS && bankAccount?.accountData?.allowDebit);
```

The bank account must be a verified business withdrawal bank account to be eligible for the Expensify Card

**Review by MariaHCD — COMMENTED** [review]
LGTM! Just one comment

File: src/libs/CardUtils.ts (hunk 11 above)
Comment (koko57) [reply src/libs/CardUtils.ts]: @MariaHCD thanks! done!

**Review by MariaHCD — APPROVED** [review]

**Comment by dangrous** [comment]
probably just got the ping but looks like the "Reconciliation account" link isn't working properly. Not sure if this is the only issue but I am not sure it's accounting for if you don't have a reconciliation account - or an accounting integration - yet

File: src/types/onyx/ExpensifyCardSettings.ts [hunk 12]
```diff
@@ -10,6 +10,15 @@ type ExpensifyCardSettings = OnyxCommon.OnyxValueWithOfflineFeedback<{
 
     /** The total amount of cash back earned thus far */
     cashBack: number;
+
+    /** The date of the last settlement */
+    monthlySettlementDate: Date;
```
Comment (linhvovan29546) [inline src/types/onyx/ExpensifyCardSettings.ts:15]: ### Checklist:

`monthlySettlementDate` from BE is a `number` (day of month), not a `timestamp`. [#98164](https://github.com/Expensify/App/pull/98164) passed it straight into new Date(), which misread the day-of-month as epoch ms  causing [#100912](https://github.com/Expensify/App/issues/100912). Fixed by normalizing the field to `number`  instead of `new Date()` directly.
